homework-09/main.js

```diff
@@ -0,0 +1,154 @@
+// 1
+function Human(name, age, gender, nationality, country, listOfCountries) {
+  this.name = name;
+  this.age = age;
+  this.gender = gender;
+  this.nationality = nationality;
+  this.country = country;
+  this.listOfCountries = listOfCountries;
+};
+
+function greeting(greet) {
+  greet = `Hello! My name is ${this.name} I'm ${this.age}, ${this.gender}
+              I'm from ${this.country}. My nationality is ${this.nationality}
+              I'd like to visit ${this.listOfCountries} countries`;
+  return greet;
+};
+
+function wakeUp(awake) {
+  awake = `Good morning! Have a good day ${this.name}`;
+  return awake;
+};
+
+function fallAsleep(asleepy) {
+  asleepy = `You've had the productive day Good night ${this.name}`;
+  return asleepy;
+};
+
+function countriesOfDram(travel) {
+  travel = `Hello ${this.name}!  Your journey begins of ${this.listOfCountries} countries`;
+  return travel;
+};
+
+const person = new Human(
+  "Alisa",
+  24,
+  "femail",
+  "ukranian",
+  "Ukraine",
+  "Czech Republic, Poland, US, Canada"
+);
+const person1 = new Human(
+  "Bob",
+  25,
+  "male",
+  "ukranian",
+  "Ukraine",
+  "US, German ,UAE"
+);
+
+console.log(greeting.apply(person));
+console.log(wakeUp.apply(person));
+console.log(fallAsleep.apply(person));
+console.log(countriesOfDram.apply(person));
+
+console.log(greeting.apply(person1));
+console.log(wakeUp.apply(person1));
+console.log(fallAsleep.apply(person1));
+console.log(countriesOfDram.apply(person1));
+
+// 2
+let canTravel = function () {
+  if (this.gender === "male" && this.age > 18 && this.country === "Ukraine") {
+    return `${this.name} will be able to travel in half a year`;
+  } else {
+    return `${this.name } have a nice trip`;
+  }
+};
+
+Function.prototype.myBind = function (obj) {
+  const oldThis = this;
+  return () => oldThis.call(obj);
+};
+
+let f = canTravel.myBind(person1);
+console.log(f()); // Bob will be able to travel in half a year

```

Comment:
Good, but what if the function needs arguments to be provided? You code can cover the case for functions that will be called with no arguments